### DISCUSSION TITLE
Prevent exit from FORM

### DISCUSSION THREAD

**Discussion by ciobanugabriel** [filamentphp/filament#12979] [Ideas]
Hi, can you help me catch the event when a user is filling out a form and wants to exit? I want to show a modal with the options: "Are you sure? Discard changes and Save as Draft." Thank you!

**Comment by dissto** [comment]
Why not use the built in https://filamentphp.com/docs/3.x/panels/configuration#unsaved-changes-alerts

**Reply by ciobanugabriel** [reply]
Thank you! It helps for the moment but I want to replace Alert with a nice Modal that have custom Actions

**Reply by dissto** [reply]
I suppose you could check the source 
https://github.com/filamentphp/filament/blob/3.x/packages%2Fpanels%2Fresources%2Fviews%2Fcomponents%2Fpage%2Funsaved-data-changes-alert.blade.php#L29
